aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -68,6 +68,7 @@ class KernelTable_ final {
 
   void removeIfExists(TensorTypeId key, const std::string& operator_name) {
     auto num_removed = map_.erase(key);
+    (void)num_removed; // silence unused compiler warning
     assert(num_removed <= 1); // This is not a multi-map
```

Comment:
Should probably change assert to TORCH_INTERNAL_ASSERT. I think that will solved the unused variable issue as well

Some background https://github.com/pytorch/pytorch/blob/master/c10/util/Exception.h#L188

Reply:
You're right.